controllers/clustergroupupgrade_controller.go

```diff
@@ -1007,40 +1053,43 @@ func (r *ClusterGroupUpgradeReconciler) ensureBatchPlacementRule(ctx context.Con
 	})
 
 	err := r.Client.Get(ctx, client.ObjectKey{
-		Name:      prName,
+		Name:      safeName,
 		Namespace: clusterGroupUpgrade.Namespace,
 	}, foundPlacementRule)
 
 	if err != nil {
 		if errors.IsNotFound(err) {
 			err = r.Client.Create(ctx, pr)
 			if err != nil {
-				return err
+				return "", err
 			}
+			clusterGroupUpgrade.Status.SafeResourceNames[name] = safeName
 		} else {
-			return err
+			return "", err
 		}
 	} else {
 		pr.SetResourceVersion(foundPlacementRule.GetResourceVersion())
 		err = r.Client.Update(ctx, pr)
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
-
-	return nil
+	return safeName, nil
 }
 
-func (r *ClusterGroupUpgradeReconciler) newBatchPlacementRule(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, prName string) *unstructured.Unstructured {
+func (r *ClusterGroupUpgradeReconciler) newBatchPlacementRule(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, policyName, name, desiredName string) *unstructured.Unstructured {
 	u := &unstructured.Unstructured{}
 	u.Object = map[string]interface{}{
 		"metadata": map[string]interface{}{
-			"name":      prName,
+			"name":      name,
 			"namespace": clusterGroupUpgrade.Namespace,
 			"labels": map[string]interface{}{
 				"app": "openshift-cluster-group-upgrades",
 				"openshift-cluster-group-upgrades/clusterGroupUpgrade": clusterGroupUpgrade.Name,
-				"openshift-cluster-group-upgrades/forPolicy":           prName,
+				"openshift-cluster-group-upgrades/forPolicy":           policyName,
+			},
+			"annotations": map[string]interface{}{
+				utils.ResourceNameAnnotation: desiredName,
```

Comment:
This is the name the user initially wanted to set up for the resource? If so, I think I would change "ResourceNameAnnotation" to something more intuitive, maybe even "DesiredName" or "InitialName" or something similar.

Reply:
Changed it DesiredResourceNameAnnotation

Reply:
Do we need to keep "Annotation" at the end? Seems extra.